pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -791,7 +851,7 @@ void updateIdealStateOnSegmentCompletion(String realtimeTableName, String commit
 
   @VisibleForTesting
   void updateInstanceStatesForNewConsumingSegment(Map<String, Map<String, String>> instanceStatesMap,
-      @Nullable String committingSegmentName, String newSegmentName, SegmentAssignment segmentAssignment,
+      @Nullable String committingSegmentName, @Nullable String newSegmentName, SegmentAssignment segmentAssignment,
```

Comment:
Instead of making newSegmentName nullable, is it not better to NOT call this method at all if neweSegmentName is null?

Reply:
We still want to make the update for the committingSegment rt?